src/Mod/Path/PathTests/TestPathSlot.py

```diff
@@ -0,0 +1,264 @@
+        is able to call static methods within this same class.
+        """
+        # FreeCAD.Console.PrintMessage("TestPathSlot.setUpClass()\n")
+        cls._setup_test_environment()
+
+    @classmethod
+    def tearDownClass(cls):
+        """tearDownClass()...
+        This method is called prior to destruction of this test class.  Add code and objects here
+        that cleanup the test environment after the test() methods in this class have been executed.
+        This method does not have access to the class `self` reference.  This method
+        is able to call static methods within this same class.
+        """
+        # FreeCAD.Console.PrintMessage("TestPathSlot.tearDownClass()\n")
+        FreeCAD.closeDocument(FreeCAD.ActiveDocument.Name)
+
+    def setUp(self):
+        """setUp()...
+        This method is called prior to each test() method.  Add code and objects here
+        that are needed for multiple test() methods.
+        """
+        self.doc = FreeCAD.ActiveDocument
+        self.con = FreeCAD.Console
+
+    def tearDown(self):
+        """tearDown()...
+        This method is called after each test() method. Add cleanup instructions here.
+        Such cleanup instructions will likely undo those in the setUp() method.
+        """
+        pass
+
+    def _set_depths_and_heights(self, op):
+        # Set Depths
+        # set start and final depth in order to eliminate effects of stock (and its default values)
+        op.setExpression('StartDepth', None)
+        op.StartDepth.Value = 10.0
+        op.setExpression('FinalDepth', None)
+        op.FinalDepth.Value = 5.0
+        op.setExpression('StepDown', None)
+        op.StepDown.Value = 5.0
+
+        # Set Heights
+        # default values used
+
+    def _format_point(self, cmd):
+        """Accepts command dict and returns point string coordinate"""
+        x = round(cmd["X"], 2)
+        y = round(cmd["Y"], 2)
+        return "({}, {})".format(x, y)
+
+    def test00(self):
+        '''Test Slot on horizontal rectangular face: ReverseDirection=True; LayerMode=Single-pass'''
+
+        # Identify base feature(s) to be used for operation's Base Geometry
+        base = self.doc.Cut
+        for i in range(11):
+            face = "Face%d" % (i+1)
+            f = base.Shape.getElement(face)
+            if (f.Surface.Axis == FreeCAD.Vector(0,0,1) and
+                f.Orientation == 'Forward' and f.BoundBox.ZMin == 5.0):
+                break
+
+        # Instantiate a Slot operation and sets standard depths & heights
+        slot = PathSlot.Create('Slot-test00')
+        self._set_depths_and_heights(slot)
+
+        # Set Base Geometry
+        slot.Base = (base, [face])
+
+        # Set Operation
+        slot.ReverseDirection = True
+        slot.LayerMode = "Single-pass"
+
+        slot.recompute()
+        self.doc.recompute()
+        
+        pnts = list()
+        for c in slot.Path.Commands:
+            p = c.Parameters
```

Comment:
what about the `Command` ?
It seems unnecessarily roundabout testing - you know the start point and the end point, so there should be a
* `G0` to safe height
* `G0` to starting point
* `G1` in z down to operation depth
* `G1` to end point
* `G1` (or `G0`) up to safe height
There should be no `G2` or `G3` commands and every other command can be ignored, like comments. As a reader I would immediately see what is being tested.

Reply:
I don't think I understand what change you are suggesting.  
.
My current strategy is to search all the commands for the two points that should exist within the command set.  
.
Having revisited this Slot op now that I found the issue with the Linux gcc-7 build, I know I need to include depth verification in all the unit tests I am creating.  I will have to figure out how to incorporate such verification across the board in all of them.  I think I have an idea how to do it.